example/echo_event_loop.ml

```diff
@@ -0,0 +1,172 @@
+open Core
```

Comment:
Can you add a brief comment describing what this module is (e.x. how it differs from how promise.ml implements Promise.t)?

Reply:
I'll actually cut this file completely, It's very similar to the promise version. Was just some initial prototyping before trying to skin Async (https://github.com/jawline/ocamlIoUringAsyncBackend here).